src/cleanfigure.m

```diff
@@ -799,30 +845,87 @@ function simplifyLine(meta, handle, targetResolution)
   if isYlog
     yLim  = log10(yLim);
   end
+  if is3D
+      isZlog = strcmp(get(meta.gca, 'ZScale'), 'log');
```

Comment:
Nitpicking mode: 

I think `ZScale` is always present on the axis, so there is little need to put it inside `if is3D, ..., end`, you could just define it in the same way as for `X` and `Y` (that would make it more obvious to refactor a bit later on).

The same goes for `ZLim`: that one is always defined (but the values, obviously, only make sense when the axis is three-dimensional).

I think that allows to keep the flow of the code a bit more linear and hence easier to follow for future readers. (This comes at the cost that now the workspace always has those Z-variables that don't always make sense).